nextjs-13/src/app/auth/sign-in/page.tsx

```diff
@@ -1,15 +1,30 @@
-"use client"
+'use client';
 import Head from 'next/head';
-import { signIn } from "next-auth/react"
+import { redirect } from 'next/navigation';
+import { signIn, useSession } from 'next-auth/react';
 
 const AuthSignInPage = () => {
+  const { status } = useSession();
+
+  if (status === 'loading') {
```

Comment:
How about using suspense: https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming?

Reply:
Woa 👀, oke let me give it a try!

Reply:
Hi @olivierobert, I just took a look over the `Suspense` and I think we can not/ should not use it here.

As I understand, in order to toggle the `Loading` component on and off, we have to depend on the React's `Suspense`'s detection.

[The React documentation](https://react.dev/reference/react/Suspense) says that:

> Only Suspense-enabled data sources will activate the Suspense component. They include:
> - Data fetching with Suspense-enabled frameworks like [Relay](https://relay.dev/docs/guided-tour/rendering/loading-states/) and [Next.js](https://nextjs.org/docs/advanced-features/react-18)
> - Lazy-loading component code with [lazy](https://react.dev/reference/react/lazy).

Since `useSession` is not data fetching nor lazy loading component, I don't know how can we activate it with this `useSession` 🤔.

Maybe for now, `Suspense` is more suitable for use cases like "Waiting for some data to fetch" or "Waiting for some images to load".

Reply:
Understood. Good you did the research as I was unaware of these limitations. We can keep in mind when implementing the CRUD for newsletters 💡